### PR TITLE
Cherrypick/fix convert process batch response

### PR DESCRIPTION
**Pull request by ToniRamirezM**
Updates develop with the contents of v0.3.1

### CODE REVIEW

File: sequencer/finalizer.go [hunk 1]
```diff
@@ -1006,12 +1007,16 @@ func (f *finalizer) processForcedBatch(ctx context.Context, lastBatchNumberInSta
...
-				continue
+			if !errors.Is(txResponse.RomError, executor.RomErr(executor.RomError_ROM_ERROR_INVALID_RLP)) {
+				sender, err := state.GetSender(txResponse.Tx)
+				if err != nil {
+					log.Warnf("failed trying to add forced tx (%s) to worker. Error getting sender from tx, Err: %v", txResponse.TxHash, err)
+					continue
+				}
+				f.worker.AddForcedTx(txResponse.TxHash, sender)
+			} else {
+				log.Warnf("ROM_ERROR_INVALID_RLP error received from executor for forced batch %d", forcedBatch.ForcedBatchNumber)
```
Comment (agnusmor) [inline sequencer/finalizer.go:1018]: Could be is useful to log here also the txResponse.TxHash (I understand that we have a valid value on txResponse.TxHash although we got an error for this tx):

`log.Warnf("ROM_ERROR_INVALID_RLP error received from executor for forced tx (%s) in forced batch %d", txResponse.TxHash, forcedBatch.ForcedBatchNumber)
`

File: state/converters.go [hunk 2]
```diff
@@ -151,39 +151,51 @@ func (s *State) convertToProcessTransactionResponse(txs []types.Transaction, res
...
+
+				eventErr := s.eventLog.LogEvent(context.Background(), event)
+				if eventErr != nil {
+					log.Errorf("error storing payload: %v", err)
+				}
+
+				return nil, err
+			}
+		} else {
+			log.Warnf("ROM_ERROR_INVALID_RLP returned by the executor")
```
Comment (agnusmor) [inline state/converters.go:179]: Log also tx hash?

File: state/converters.go [hunk 3]
```diff
@@ -151,39 +151,51 @@ func (s *State) convertToProcessTransactionResponse(txs []types.Transaction, res
...
-				log.Errorf("error storing payload: %v", err)
+				timestamp := time.Now()
+				log.Errorf("error decoding rlp returned by executor %v at %v", err, timestamp)
+
+				event := &event.Event{
+					ReceivedAt: timestamp,
+					Source:     event.Source_Node,
+					Level:      event.Level_Error,
+					EventID:    event.EventID_ExecutorRLPError,
+					Json:       string(response.GetRlpTx()),
```
Comment (agnusmor) [inline state/converters.go:168]: Log also tx hash?

File: sequencer/finalizer.go (hunk 1 above)
Comment (ToniRamirezM) [reply sequencer/finalizer.go:1018]: If the batchdata cannot be decoded into RLP it is also not possible to calculate its hash. That's why in other point of the code we log the forced batch number, to know where this tx came from.

File: state/converters.go (hunk 2 above)
Comment (ToniRamirezM) [reply state/converters.go:179]: Same comment as before.

File: state/converters.go (hunk 3 above)
Comment (ToniRamirezM) [reply state/converters.go:168]: And again :)

**Review by agnusmor — APPROVED** [review]

**Pull request merged**